src/core/src/package_managers/AptitudePackageManager.py

```diff
@@ -524,3 +540,27 @@ def update_os_patch_configuration_sub_setting(self, patch_configuration_sub_sett
     def do_processes_require_restart(self):
         """Defaulting this for Apt"""
         return False
+
+    def is_reboot_pending(self):
+        """Pro client implementation for reboot pending/"""
+        # If pre-requisite is False, raise exception to fall back
+        # to existing way in package_manager
+        if not self.__pro_client_prereq_met:
+            self.composite_logger.log_debug("Pro client pre-requisite not met.")
+            return False, False
+        return self.ubuntu_pro_client.is_reboot_pending()
+
+    def __is_pro_client_prereq_met(self):
+        """check required conditions to use pro client"""
+        if Constants.PRO_CLIENT_ENABLED \
+                and self.__get_os_major_version() <= self.maximum_os_version_supported \
+                and self.ubuntu_pro_client.is_pro_working():
+            return True
+        else:
+            return False
+
+    def __get_os_major_version(self):
+        """ get the os major version"""
```

Comment:
Good to add a sample of what linux_distribution() would return

Also nit: """get the OS major version"""